### ISSUE TITLE
Submit handler won't fire even after validations pass

### ISSUE THREAD

**Issue by paambaati** [react-hook-form/react-hook-form#2191]
**Describe the bug**
In a form with a `validate` prop set on `register`, the submit handler won't fire after the user fixes a validation error.

**To Reproduce**
1. Go to https://codesandbox.io/s/smoosh-sunset-t8q22?file=/src/App.js
2. Enter some numeric value for the Count 1 field, and then some numeric value for the Weight 1 field, and see the form getting submitted (this is expected).
3. Now refresh the page, and on refresh, press submit without entering anything. Notice validation failures in the console (this is expected).
4. Now without refreshing, enter some numeric value for Count 1, and some numeric value for Count 2, and click on Submit.
5. Notice that validation succeeds (see console, and this is expected) **but** the form's submit handler won't fire. This is the problem, as I'd expect the form to submit now that all validations pass.

**Codesandbox link**
https://codesandbox.io/s/smoosh-sunset-t8q22?file=/src/App.js

**Expected behavior**
On submitting the form after all validation errors are fixed, the submit handler should be fired.

**Desktop:**
 - OS: macOS Mojave 10.14.6
 - Browser Chrome 85.0.4183.16 (Official Build) dev (64-bit)
 - Version `react-hook-form` v6.0.4

**Additional context**
Might be somewhat related to https://github.com/react-hook-form/react-hook-form/issues/2180, but I'm not a 100% sure.

**Quick note about the validation logic**

1. `atLeast1Count` - The user is expected to enter input for at least one of the 'Count' fields.
2. `requiredWeight` - For every 'Count' field that has a value > 0, the corresponding 'Weight' field should also have a value.

**Comment by bluebill1049** [comment]
can you reproduce into a simpler codesandbox, if the validate is not working a simple codesandbox can reproduce the problem? in fact, I will do a codesandbox to prove it's working or not.

**Comment by bluebill1049** [comment]
nvm, i think i see what you try to proof now.

**Comment by bluebill1049** [comment]
This is not going to work you register at roof without any `name`. I think you try to share the validation logic, you should build those into a logic file and share the validator 

eg:
`register('test', youSharedValidation)`

**Comment by paambaati** [comment]
@bluebill1049 I've updated the CSB by making 2 changes - 

1. Setting a `name` on the component and on the `register` call.
2. Moving validation logic out to a function.

Now, the validations don't kick in at all and instead, the form gets submitted immediately.

**Comment by bluebill1049** [comment]
I think we can close this issue now: https://codesandbox.io/s/wandering-cdn-m61er?file=/src/ComplexInput.jsx

take a look at the codesandbox, it should resolve your problem.

**Comment by paambaati** [comment]
@bluebill1049 Thanks for the CSB, I can see it works. But I made a fork of it to add some of my changes (only in validation), and it still doesn't work 😓.

See https://codesandbox.io/s/icy-snowflake-3dzgo?file=/src/ComplexInput.jsx

**Steps to reproduce**

1. Submit form without any inputs. Notice it won't submit (expected, as validation fails).
2. Now enter any input in Count 1 field.
3. Then enter any input in Count 2 field.
4. Now submit form. See that both validations pass (notice console), but the form still won't submit.

**Differences from your CSB**

1. Added validation logic that tests for -

	a. At least 1 count field must be entered.
	b. For every count field that has a value, the corresponding weight field must have a value too.

So the key difference in my change is that validations dynamically depend on `getValues()`.

What is confusing for me is that both validations fail and pass correctly based on inputs, but the form won't submit when validation has failed at least once. On a refreshed page with all inputs entered correctly, the form submits without any issues.
